statsmodels/tsa/stattools.py

```diff
@@ -268,6 +268,9 @@ def adfuller(
     store = bool_like(store, "store")
     regresults = bool_like(regresults, "regresults")
 
+    if x.max() == x.min():
+        raise ValueError("Invalid input, residual variance is zero")
```

Comment:
I think this should have a more specific message, e.g.
`ValueError("Invalid data, x is constant")`

Reply:
Hi, just updated that message to report "x is constant"